App/ViewModels/FileSyncListViewModel.cs

```diff
@@ -0,0 +1,253 @@
+    public partial string NewSessionLocalPath { get; set; } = "";
+
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(NewSessionCreateEnabled))]
+    public partial bool NewSessionLocalPathDialogOpen { get; set; } = false;
+
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(NewSessionCreateEnabled))]
+    public partial string NewSessionRemoteName { get; set; } = "";
+
+    [ObservableProperty]
+    [NotifyPropertyChangedFor(nameof(NewSessionCreateEnabled))]
+    public partial string NewSessionRemotePath { get; set; } = "";
+    // TODO: NewSessionRemotePathDialogOpen for remote path
+
+    public bool NewSessionCreateEnabled
+    {
+        get
+        {
+            if (string.IsNullOrWhiteSpace(NewSessionLocalPath)) return false;
+            if (NewSessionLocalPathDialogOpen) return false;
+            if (string.IsNullOrWhiteSpace(NewSessionRemoteName)) return false;
+            if (string.IsNullOrWhiteSpace(NewSessionRemotePath)) return false;
+            return true;
+        }
+    }
+
+    // TODO: this could definitely be improved
+    public bool ShowUnavailable => UnavailableMessage != null;
+    public bool ShowLoading => Loading && UnavailableMessage == null && Error == null;
+    public bool ShowError => UnavailableMessage == null && Error != null;
+    public bool ShowSessions => !Loading && UnavailableMessage == null && Error == null;
+
+    public FileSyncListViewModel(ISyncSessionController syncSessionController, IRpcController rpcController,
+        ICredentialManager credentialManager)
+    {
+        _syncSessionController = syncSessionController;
+        _rpcController = rpcController;
+        _credentialManager = credentialManager;
+
+        Sessions =
+        [
+            new SyncSessionModel(@"C:\Users\dean\git\coder-desktop-windows", "pog", "~/repos/coder-desktop-windows",
+                SyncSessionStatusCategory.Ok, "Watching", "Some description", []),
+            new SyncSessionModel(@"C:\Users\dean\git\coder", "pog", "~/coder", SyncSessionStatusCategory.Paused,
+                "Paused",
+                "Some description", []),
+            new SyncSessionModel(@"C:\Users\dean\git\coder", "pog", "~/coder", SyncSessionStatusCategory.Conflicts,
+                "Conflicts", "Some description", []),
+            new SyncSessionModel(@"C:\Users\dean\git\coder", "pog", "~/coder", SyncSessionStatusCategory.Error,
+                "Halted on root emptied", "Some description", []),
+            new SyncSessionModel(@"C:\Users\dean\git\coder", "pog", "~/coder", SyncSessionStatusCategory.Unknown,
+                "Unknown", "Some description", []),
+            new SyncSessionModel(@"C:\Users\dean\git\coder", "pog", "~/coder", SyncSessionStatusCategory.Working,
+                "Reconciling", "Some description", []),
+        ];
+    }
+
+    public void Initialize(DispatcherQueue dispatcherQueue)
+    {
+        _dispatcherQueue = dispatcherQueue;
```

Comment:
do we also need to check `HasThreadAccess`?

Reply:
This is only being called in one place in a main thread, I could add a check for that and throw an exception if you want.

Reply:
Check and throw is good. Want to avoid little landmines on the public interface of this class.